src/ArgumentParser.cpp

```diff
@@ -0,0 +1,91 @@
+#include "ArgumentParser.h"
+
+vector<pair<int, int>> ArgumentParser::m_fds;
+
+void ArgumentParser::resetFds() {
+    for (const auto &pair: m_fds) {
+        if (dup2(pair.first, pair.second) == -1) {
+            throw std::runtime_error("Failed to restore fd");
+        }
+        if (close(pair.first) == -1) {
+            throw std::runtime_error("Failed to close fd");
+        }
+    }
+    m_fds.clear();
+}
+
+string ArgumentParser::resolveCommandPath(const string &command) {
+    // Check if the command is already a full path
+    if (command.find('/') == 0 && access(command.c_str(), X_OK) == 0) {
```

Comment:
The command can be a path to a file even if it doesn't start with "/", it could be a relative path. 
Also, consider using `std::filesystem::exists` or an "is_file" function instead of access which is old-fashioned